cmake/EnableCsharp.cmake

```diff
@@ -5,16 +5,14 @@ if(WIN32)
 else()
   # for other platforms we currently use mono
   find_program(MONO_EXECUTABLE mono)
-  find_program(MCS_EXECUTABLE dmcs)
+  find_program(MCS_EXECUTABLE mcs)
 
   if (NOT MCS_EXECUTABLE)
-    find_program(MCS_EXECUTABLE mcs)
+    find_program(MCS_EXECUTABLE dmcs)
   endif()
 
-  set(MONO_FOUND FALSE CACHE INTERNAL "")
-
   if (NOT MCS_EXECUTABLE)
-    find_program(MCS_EXECUTABLE mcs)
+    set(MONO_FOUND FALSE CACHE INTERNAL "")
   endif()
 
   if (MONO_EXECUTABLE AND MCS_EXECUTABLE)
```

Comment:
Is this `if`-block obsolete now? I think the whole block can and should be removed

Reply:
Can I go one step further: could we just only check for `mcs` and bail if it isn't there? :)